modal-js/src/dict.ts

```diff
@@ -0,0 +1,277 @@
+import {
+  ObjectCreationType,
+  DictEntry,
+  DictContentsRequest,
+} from "../proto/modal_proto/api";
+import type { DeleteOptions, EphemeralOptions, LookupOptions } from "./app";
+import { client } from "./client";
+import { environmentName } from "./config";
+import { InvalidError, KeyError, RequestSizeError } from "./errors";
+import { dumps, loads } from "./pickle";
+import { ClientError } from "nice-grpc";
+
+// From: modal/_object.py
+const ephemeralObjectHeartbeatSleep = 300_000; // 300 seconds
+
+/** Options to configure a `Dict.put()` operation. */
+export type DictPutOptions = {
+  /** Skip adding the key if it already exists. */
+  skipIfExists?: boolean;
+};
+
+// Dict is a distributed dictionary for key-value storage in Modal apps.
+export type DictKey = string | Uint8Array;
+export type DictUpdateItems<K extends DictKey, V> = K extends string
```

Comment:
I think the `V` is already restricted to `string | Uint8Array` as well?

Reply:
No `V` can be `any` in the current implementation! We're just limiting keys.